exasol_script_languages_container_ci_setup/lib/deployer.py

```diff
@@ -0,0 +1,212 @@
+            'Parameters': parameter_values,
+            'Capabilities': capabilities,
+            'Description': description,
+            'Tags': tags,
+        }
+
+        # don't set these arguments if not specified to use existing values
+        if role_arn is not None:
+            kwargs['RoleARN'] = role_arn
+        if notification_arns is not None:
+            kwargs['NotificationARNs'] = notification_arns
+        try:
+            resp = self._client.create_change_set(**kwargs)
+            return ChangeSetResult(resp["Id"], changeset_type)
+        except Exception as ex:
+            LOG.debug("Unable to create changeset", exc_info=ex)
+            raise ex
+
+    def wait_for_changeset(self, changeset_id, stack_name):
+        """
+        Waits until the changeset creation completes
+
+        :param changeset_id: ID or name of the changeset
+        :param stack_name:   Stack name
+        :return: Latest status of the create-change-set operation
+        """
+        LOG.info("Waiting for changeset to be created..")
+
+        # Wait for changeset to be created
+        waiter = self._client.get_waiter("change_set_create_complete")
+        # Poll every 5 seconds. Changeset creation should be fast
+        waiter_config = {'Delay': 5}
+        try:
+            waiter.wait(ChangeSetName=changeset_id, StackName=stack_name,
+                        WaiterConfig=waiter_config)
+        except botocore.exceptions.WaiterError as ex:
+            LOG.debug("Create changeset waiter exception", exc_info=ex)
+
+            resp = ex.last_response
+            status = resp["Status"]
+            reason = resp["StatusReason"]
+
+            raise RuntimeError("Failed to create the changeset: {0} "
+                               "Status: {1}. Reason: {2}"
+                               .format(ex, status, reason)) from ex
+
+    def execute_changeset(self, changeset_id, stack_name,
+                          disable_rollback=False):
+        """
+        Calls CloudFormation to execute changeset
+
+        :param changeset_id: ID of the changeset
+        :param stack_name: Name or ID of the stack
+        :param disable_rollback: Disable rollback of all resource changes
+        :return: Response from execute-change-set call
+        """
+        return self._client.execute_change_set(
+            ChangeSetName=changeset_id,
+            StackName=stack_name,
+            DisableRollback=disable_rollback)
+
+    def wait_for_execute(self, stack_name, changeset_type):
+
+        LOG.info("Waiting for stack create/update to complete\n")
+
+        # Pick the right waiter
+        if changeset_type == "CREATE":
+            waiter = self._client.get_waiter("stack_create_complete")
+        elif changeset_type == "UPDATE":
+            waiter = self._client.get_waiter("stack_update_complete")
+        else:
+            raise RuntimeError("Invalid changeset type {0}"
+                               .format(changeset_type))
```

Comment:
```suggestion
        # Pick the right waiter
        mapping = { "CREATE" :"stack_create_complete", "UPDATE": "stack_update_complete" }
        try:
            waiter = self._client.get_waiter(mapping[changeset_type])
        except KeyError as ex:
            raise RuntimeError(f"Invalid changeset type {ex}") from ex
```

Reply:
I would prefer to avoid further modifications of the AWS code, as it will make future updates more complicated...